pkg/daemon/config.go

```diff
@@ -0,0 +1,87 @@
+package daemon
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/golang/glog"
+
+	ptpv1 "github.com/openshift/ptp-operator/pkg/apis/ptp/v1"
+)
+
+// LinuxPTPUpdate controls whether to update linuxPTP conf
+// and contains linuxPTP conf to be updated. It's rendered
+// and passed to linuxptp instance by daemon.
+type LinuxPTPConfUpdate struct {
+	UpdateCh               chan bool
+	NodeProfiles           []ptpv1.PtpProfile
+	appliedNodeProfileJson []byte
+	defaultPTP4lConfig     []byte
+}
+
+func NewLinuxPTPConfUpdate() (*LinuxPTPConfUpdate, error) {
+	if _, err := os.Stat(PTP4L_CONF_FILE_PATH); err != nil {
+		if os.IsNotExist(err) {
+			return nil, fmt.Errorf("ptp.conf file doesn't exist")
+		} else {
+			return nil, fmt.Errorf("unknow error searching for the %s file: %v", PTP4L_CONF_FILE_PATH, err)
+		}
+	}
+
+	defaultPTP4lConfig, err := ioutil.ReadFile(PTP4L_CONF_FILE_PATH)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read %s: %v", PTP4L_CONF_FILE_PATH, err)
+	}
+
+	return &LinuxPTPConfUpdate{UpdateCh: make(chan bool), defaultPTP4lConfig: defaultPTP4lConfig}, nil
+}
+
+func (l *LinuxPTPConfUpdate) UpdateConfig(nodeProfilesJson []byte) error {
+	if string(l.appliedNodeProfileJson) == string(nodeProfilesJson) {
+		return nil
+	}
+
+	if nodeProfiles, ok := tryToLoadOldConfig(nodeProfilesJson); ok {
```

Comment:
In which case would this be required? when operator and operand have mis-matched versions? like during upgrade period?

Reply:
yes this can break if the operator is updated and also the operand but the reconcile on the policy takes more time to change to the new struct.

but let me change and first call the new method it will make the parse a bit faster as only at the beginning we can have the issue after that it should always use the new struct